### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by mikehearn**
Some inspector-driven code cleanups in core and node. Dead code elimination and other misc inspections applied.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/crypto/EdDSAKeyFactory.kt [hunk 1]
```diff
@@ -7,6 +7,4 @@ import java.security.KeyFactory
  * This is required as a [SignatureScheme] requires a [java.security.KeyFactory] property, but i2p has
  * its own KeyFactory for EdDSA, thus this actually a Proxy Pattern over i2p's KeyFactory.
  */
-class EdDSAKeyFactory : KeyFactory {
-    constructor() : super(net.i2p.crypto.eddsa.KeyFactory(), null, "EDDSA_ED25519_SHA512")
-}
+class EdDSAKeyFactory : KeyFactory(net.i2p.crypto.eddsa.KeyFactory(), null, "EDDSA_ED25519_SHA512")
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/crypto/EdDSAKeyFactory.kt]: This is can an `object`

File: node/src/main/kotlin/net/corda/node/services/persistence/DBTransactionMappingStorage.kt [hunk 2]
```diff
@@ -42,10 +42,11 @@ class DBTransactionMappingStorage : StateMachineRecordedTransactionMappingStorag
         }
     }
 
-    private val mutex = ThreadBox(object {
+    private class InnerState {
         val stateMachineTransactionMap = TransactionMappingsMap()
-        val updates = PublishSubject.create<StateMachineTransactionMapping>()
-    })
+        val updates = PublishSubject.create<StateMachineTransactionMapping>()!!
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/persistence/DBTransactionMappingStorage.kt]: Give the explicit type and then get rid of the `<>` and `!!`

File: node/src/main/kotlin/net/corda/node/services/transactions/DistributedImmutableMap.kt [hunk 3]
```diff
@@ -43,11 +43,9 @@ class DistributedImmutableMap<K : Any, V : Any>(val db: Database, tableName: Str
 
     /** Gets a value for the given [Commands.Get.key] */
     fun get(commit: Commit<Commands.Get<K, V>>): V? {
-        try {
+        commit.use {
             val key = commit.operation().key
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/transactions/DistributedImmutableMap.kt]: it

File: node/src/main/kotlin/net/corda/node/services/vault/VaultSoftLockManager.kt [hunk 4]
```diff
@@ -19,12 +19,12 @@ class VaultSoftLockManager(val vault: VaultService, smm: StateMachineManager) {
     private val trackingFlowIds: MutableSet<UUID> = Collections.synchronizedSet(HashSet())
 
     init {
-        smm.changes.subscribe { change ->
-            if (change.addOrRemove == AddOrRemove.REMOVE && trackingFlowIds.contains(change.id.uuid)) {
-                log.trace { "${change.addOrRemove} Flow name ${change.logic.javaClass} with id ${change.id}" }
-                unregisterSoftLocks(change.id, change.logic)
+        smm.changes.subscribe { (logic, addOrRemove, id) ->
+            if (addOrRemove == AddOrRemove.REMOVE && trackingFlowIds.contains(id.uuid)) {
```
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/vault/VaultSoftLockManager.kt]: `in` instead of contains

**Comment by mikehearn** [comment]
Merged (CI issue is unrelated, some RAM problem)

**Pull request closed without merging**